docs(observability): document apollo.router.cache.redis.reconnection and .unresponsive metrics (#9306)
apollographql/router: docs/source/routing/observability/router-telemetry-otel/enabling-telemetry/standard-instruments.mdx

- `apollo.router.cache.redis.commands_executed` - Total number of Redis commands executed
- `apollo.router.cache.redis.redelivery_count` - Number of Redis command redeliveries due to connection issues
- `apollo.router.cache.redis.errors` - Number of Redis errors by error type and cache kind
- `apollo.router.cache.redis.reconnection` - Number of reconnect events raised by the Redis client when a Redis server requires client reconnection.

> - `apollo.router.cache.redis.unresponsive` - Number of unresponsive events raised by the Redis client when a Redis server becomes unresponsive.

Reviewer: How about this to match the style of the other metrics?
```suggestion
  - `apollo.router.cache.redis.reconnection` - Number of Redis client reconnections
  - `apollo.router.cache.redis.unresponsive` - Number of times a Redis server became unresponsive
```
